nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/TransformOpValidation.java

```diff
@@ -1986,4 +1987,95 @@ public void testLogSumExp2(){
                     .gradientCheck(true));
         }
     }
+
+
+    @Test
+    public void testCRELU(){
+
+        Nd4j.getRandom().setSeed(12345);
+        INDArray inputArr = Nd4j.rand(DataType.DOUBLE, 2, 2);
+        SameDiff sd = SameDiff.create();
+        SDVariable in = sd.var(inputArr);
+
+        SDVariable crelu = new CReLU(sd,in).outputVariable();
+        INDArray expected = Nd4j.concat(1, Nd4j.nn.relu(inputArr, 0), Nd4j.nn.relu(inputArr.neg(), 0));
+
+
+        OpValidation.validate(new TestCase(sd)
+                .expectedOutput("crelu", expected)
+                .gradientCheck(true)
+        );
+    }
+
+    @Test
+    public void testClipByAvgNorm(){
+
+        Nd4j.getRandom().setSeed(12345);
+        INDArray inputArr = Nd4j.rand(DataType.DOUBLE, 2,2,2);
+        SameDiff sd = SameDiff.create();
+        SDVariable in = sd.var(inputArr);
+        SDVariable out = new ClipByAvgNorm(sd,in, 1e-2,0,1,2).outputVariable();
+        SDVariable expected = sd.math.clipByNorm(in, 1e-2,0,1,2).mul(inputArr.length());
+
+        out.markAsLoss();
+
+
+        OpValidation.validate(new TestCase(sd)
+                .expectedOutput("clipbyavgnorm", expected.eval())
+                .gradientCheck(false)
+        );
+    }
+
+
+    @Test
+    public void testDepthwiseConv2D(){
+
+        int bS = 10;
+
+        int kernelHeight = 2;
+        int kernelWidth = 2;
+        int strideHeight = 2;
+        int strideWidth = 2;
+        int inChannels = 10;
+        int outChannels = 5;
+        Nd4j.getRandom().setSeed(12345);
+        SameDiff sd = SameDiff.create();
+        SDVariable in = sd.var("in", Nd4j.rand(bS, inChannels, 28,28));
+        SDVariable weights = sd.var("weights", Nd4j.rand(DataType.FLOAT, kernelHeight, kernelWidth, inChannels, outChannels));
+        SDVariable bias = sd.var("bias", Nd4j.rand(DataType.FLOAT, outChannels));
+
+        ArrayList<SDVariable> vars_for_grads = new ArrayList<SDVariable>();
+        vars_for_grads.add(in);
+        vars_for_grads.add(weights);
+        vars_for_grads.add(bias);
+
+
+        List<SDVariable> grads = new DepthwiseConv2D(sd,in, weights, null, Conv2DConfig.builder()
+                .kH(kernelHeight)
+                .kW(kernelWidth)
+                .sH(strideHeight)
+                .sW(strideWidth)
+                .dataFormat("NCHW")
+                .build()).doDiff(vars_for_grads);
+        System.out.println(grads.get(0).eval());
+//        java.lang.IllegalStateException: Could not find descriptor for op: depthwise_conv2d_bp] - class: org.nd4j.linalg.api.ops.impl.layers.convolution.DepthwiseConv2DBp
```

Comment:
That comment: due to bad op name `depthwise_conv2d_bp]` not `depthwise_conv2d_bp`
Also just use sd.cnn.depthWiseConv2d method instead of creating manually.